### PR TITLE
Adds e2e

### PR DESCRIPTION
**Pull request by igorp1**
__What__

This PR adds the initial setup for us to be able to tests our components end-to-end, from integration to platform requests. It cleans up some of the documentation we have around to reflect most recent updates. This is a starting point but there'll be more to do in later PRs:

- testing the other 2 integration types
- including other relevant components in the examples and write e2e test for them as well.

**Example on localhost:4000** 

https://github.com/duffelhq/duffel-components/assets/2934495/86699047-7bed-40d9-bf76-7112d3b32317 

**Example on api.duffel.com** 

https://github.com/duffelhq/duffel-components/assets/2934495/26fc5875-ee21-4054-9c91-35f0ed02cb73

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,67 +2,152 @@
...
+     passengers={[ ... ]}
+     onPayloadReady={console.log}
+   />
+   ```
 
-   - [Find it on src/examples/client-side](src/examples/client-side)
+### Integrating HTML custom element
 
-2. **full-stack** | This example loads a local version of the components using a `script` tag. It will search and retrieve an offer from the Duffel API and render the ancillaries component custom element with its `offer_id` and `client_key`. As soon as a service modal is closed it will create a test order with the given payload.
+If you are not using React but still in a node environemnt you cna
```
Comment (andrejak) [inline README.md:40]:
```suggestion
If you are not using React but still in a node environemnt you can
```

Comment (andrejak) [reply README.md:40]: Unfinished sentence?

File: README.md [hunk 2]
```diff
@@ -2,67 +2,152 @@
...
+## FAQ
+
+### Are there integration guides?
+
+- [Integrating the ancillaries component to your booking flow](https://duffel.com/docs/guides/ancillaries-component).
+- [Collecting payments with Duffel Payments](https://duffel.com/docs/guides/collecting-customer-card-payments)
+
+More guides are coming soon.
+
+The `examples` folder is a great way to get started quickly and see a fully functioning examples for every component.
```
Comment (andrejak) [inline README.md:144]:
```suggestion
The `examples` folder is a great way to get started quickly and see fully functioning examples for every component.
```

File: e2e/start.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+## Build the e2e script
+esbuild e2e/index.e2e.ts --outfile=e2e/dist/index.e2e.cjs --format=cjs --platform=node --bundle --tsconfig=scripts.tsconfig.json
+
+# Test loading the component from our CDN
+## Run `using-cdn` example
+DUFFEL_API_URL=$DUFFEL_API_URL DUFFEL_API_TOKEN=$DUFFEL_API_TOKEN node examples/using-cdn/server.mjs &
+FULL_STACK_WITH_CDN_PID=$!
+echo -e "\n'using-cdn' example running on process $FULL_STACK_WITH_CDN_PID"
+
+## Run the test agasint i
```
Comment (andrejak) [inline e2e/start.sh:20]:
```suggestion
## Run the test against it
```

**Review by andrejak — COMMENTED** [review]
I get an error when I try to run the e2e tests, is there a set up step I'm missing? (I've got platform running, some creds in .env.local, and I can visit `https://localhost:4000/` in the browser)

Screenshot 2024-01-08 at 09 18 38

File: e2e/start.sh (hunk 3 above)
Comment (igorp1) [reply e2e/start.sh:20]: Fixed!

File: examples/using-cdn/README.md [hunk 4]
```diff
@@ -18,7 +18,7 @@ DUFFEL_API_TOKEN=test_duffel_dev_rw # development (for Duffel engineers only)
 
 # The url for the component CDN.
 # This is used to load both the styles an
-# COMPONENT_CDN=https://assets.duffel.com/components/ancillaries/VERSION # production
+# COMPONENT_CDN=w # production
```
[the remark this answers is not in the document]
Comment (igorp1) [reply examples/using-cdn/README.md:21]: it was, fixed it!

File: README.md (hunk 1 above)
Comment (igorp1) [reply README.md:40]: Missing colon at the end... updated!

Screenshot 2024-01-08 at 9 33 40 AM

File: README.md (hunk 2 above)
Comment (igorp1) [reply README.md:144]: Fixed!

**Comment by igorp1** [comment]
> I get an error when I try to run the e2e tests, is there a set up step I'm missing?

There's a few different parts to the setup and how you'll do it will depend on what you have on your `.env.local`. Let's sit together for a bit and walk through it together so I have a better idea of what I should document and where. But easiest setup is to run it against production. So `DUFFEL_API_TOKEN=duffel_test_..` and `DUFFEL_API_URL=https://api.duffel.com` should be on your `.env.local`.

**Review by andrejak — APPROVED** [review]
Test completed with no issues with that set up :)

**Pull request merged**